third_party/blink/renderer/platform/wtf/text/text_encoding.cc

```diff
@@ -62,8 +62,28 @@ std::string TextEncoding::Encode(const String& string,
   if (string.empty())
     return std::string();
 
-  recordreplay::Assert("[RUN-1350-1765] TextEncoding::Encode %u %s %d", string.length(), GetName(),
-                       (int)handling);
+  // For RUN-1350-1797 (add text contents to assertion.)
+  char buf1[16] = {0};
+  const char* sep = "";
+  char buf2[16] = {0};
+
+  // This won't play well with UTF8 strings with non-singlebyte codepoints.
```

Comment:
When doing byte-wise things that won't work with `Utf`, why not use `Ascii` instead?

Reply:
Well, there may be many strings that are unicode but largely ascii in content (with some higher encoded chars).  I'm hoping (1) that non-ascii strings are rare via `TextEncode` and that (2) if we do get some that there's a good chance the start and end of them are ascii.

If neither of those are true we'll get some junk assert strings out.